### PR TITLE
[OSGi] Enable Swagger

### PR DESCRIPTION
**Pull request by neykov**
~~Note that this PR bumps the guava version, so might need wider community discussion!~~

### CODE REVIEW

**Comment by ahgittin** [comment]
Apart from guava and felix version bumps this looks good.  Felix version bump seems fine.  Guava however...

Have you checked what from 16 (jclouds) and 17 (brooklyn) is incompatible with 18 (swagger) -- or even 19 (latest) ?

Alternatively it looks like the deps in the `rest-swagger/pom.xml` pointing at brooklyn are never used.  So maybe we could remove them altogether and in the OSGi build allow it to take the guava version it wants even if different to the version brooklyn uses.  (This strategy is probably more important for jclouds -- and if we had jclouds getting the guava bundle version it wants I'd be relaxed about Brooklyn taking any version.)

**Comment by ahgittin** [comment]
test failure looks related:

```
2016-03-11 08:40:51,463 INFO  TESTNG FAILED: "Surefire test" - org.apache.brooklyn.core.mgmt.osgi.OsgiVersionMoreEntityTest.testMoreEntitiesV2FailsWithoutBasicTestOsgiEntitiesBundle() finished in 47 ms
java.lang.AssertionError: Missing expected text in error: org.apache.brooklyn.util.exceptions.CompoundRuntimeException: Unable to instantiate org.apache.brooklyn.test.osgi.entities.more.MoreEntity; 2 errors including: Transformer for java-type-name gave an error creating this plan: Unable to load org.apache.brooklyn.test.osgi.entities.more.MoreEntity from [OSGi:org.apache.brooklyn.test.osgi.entities.more.MoreEntity:0.1.0[[CatalogBundleDto{symbolicName=null, version=null, url=classpath:/brooklyn/osgi/brooklyn-test-osgi-more-entities_0.2.0.jar}]]]: Unable to resolve class org.apache.brooklyn.test.osgi.entities.more.MoreEntity in CatalogBundleDto{symbolicName=null, version=null, url=classpath:/brooklyn/osgi/brooklyn-test-osgi-more-entities_0.2.0.jar}: ClassNotFoundException: org.apache.brooklyn.test.osgi.entities.more.MoreEntity: BundleException: Unable to resolve org.apache.brooklyn.test.resources.osgi.brooklyn-test-osgi-more-entities [44](R 44.0): missing requirement [org.apache.brooklyn.test.resources.osgi.brooklyn-test-osgi-more-entities [44](R 44.0)] osgi.wiring.package; (osgi.wiring.package=org.apache.brooklyn.test.osgi.entities) Unresolved requirements: [[org.apache.brooklyn.test.resources.osgi.brooklyn-test-osgi-more-entities [44](R 44.0)] osgi.wiring.package; (osgi.wiring.package=org.apache.brooklyn.test.osgi.entities)]: ClassNotFoundException: org.apache.brooklyn.test.osgi.entities.more.MoreEntity expected [true] but found [false]
	at org.apache.brooklyn.core.mgmt.osgi.OsgiVersionMoreEntityTest.testMoreEntitiesV2FailsWithoutBasicTestOsgiEntitiesBundle(OsgiVersionMoreEntityTest.java:285)
```

**Comment by ahgittin** [comment]
simple to fix, i guess the felix error messages have changed.

change `OsgiVersionMoreEntityTest:282` from

        try {
            Entity me = addItemFromCatalog(c2);
            Assert.fail("Should have failed, with unresolved dependency; instead got "+me);
        } catch (Exception e) {
            Assert.assertTrue(e.toString().toLowerCase().contains("unresolved constraint"), "Missing expected text in error: "+e);
            Assert.assertTrue(e.toString().toLowerCase().contains("wiring.package"), "Missing expected text in error: "+e);
            Assert.assertTrue(e.toString().toLowerCase().contains("org.apache.brooklyn.test.osgi.entities"), "Missing expected text in error: "+e);
        }

to
        
        try {
            Entity me = addItemFromCatalog(c2);
            Asserts.shouldHaveFailedPreviously("Should have failed, with unresolved dependency; instead got: "+me);
        } catch (Exception e) {
            Asserts.expectedFailureContainsIgnoreCase(e, "unresolved", "wiring.package", "org.apache.brooklyn.test.osgi.entities");
        }

**Comment by neykov** [comment]
Had the same failure this morning, but after pulling latest changes and rebuilding it went away.

**Comment by neykov** [comment]
Forcing a rebuild.

**Comment by neykov** [comment]
Re guava version change - haven't looked at the specific API changes between versions, but would expect that they would lead to a compile error if something used by Brooklyn changed. Will take a look at the release notes to confirm.

I suggest that we follow the jclouds lead and stick to the least common denominator API between guava versions. Then we can leave the v17 dependency in the pom, but load 16-18 in Karaf, as we see fit.

**Comment by neykov** [comment]
Loading both guava 17 & guava 18 worked out fine, sticking to this solution.

Leaving brooklyn to depend on 17.

File: utils/rest-swagger/pom.xml [hunk 1]
```diff
@@ -85,80 +76,19 @@
...
-                                </includes>
-                            </artifactSet>
-                            <finalName>${project.artifactId}-${project.version}-with-swagger</finalName>
-                            <outputFile>${project.build.directory}/${project.artifactId}-shaded-${project.version}.jar</outputFile>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>-->
-            <plugin>
-                <groupId>org.apache.felix</groupId>
```
Comment (ahgittin) [inline utils/rest-swagger/pom.xml]: nice that this can be removed now

**Comment by ahgittin** [comment]
is guava 18 included in the osgi build?  maybe worth a check it's not being too smart and trying to fetch that from the internet if not in the local maven cache.

alternatively does the swagger stuff work w the same version as brooklyn?

apart from that looks good

**Comment by ahgittin** [comment]
this conflicts w #55 but it's easy to resolve; done so in https://github.com/ahgittin/brooklyn-server/commit/b49bb07cd35eaba6a5762de1c3f988d9239de613 (which i will use when we merge this)

**Comment by neykov** [comment]
Yes, guava is included and swagger works with the newer version.

**Pull request merged**